text/204-golang-bindings.md

```diff
@@ -0,0 +1,762 @@
+   GetInterval() cdk.Duration;
+   GetRetries() int
+   GetStartPeriod() cdk.Duration;
+   GetTimeout() cdk.Duration;
+}
+
+type HealthCheck struct {
+    Command []string
+    Interval cdk.Duration;
+    Retries int
+    StartPeriod cdk.Duration;
+    Timeout cdk.Duration;
+}
+
+// See NOTE below
+func (h HealthCheck) GetCommand() []string {
+    return append([]string{}, h.command...)
+}
+
+func (h HealthCheck) GetRetries() int {
+    return h.retries
+}
+
+func (h HealthCheck) GetInterval() cdk.Duration {
+    return h.interval
+}
+
+func (h HealthCheck) GetStartPeriod() cdk.Duration {
+    return h.startPeriod
+}
+
+func (h HealthCheck) GetTimeout() cdk.Duration {
+    return h.timeout
+}
+
+func renderHealthCheck(hc HealthCheck) cfntaskdefinition.HealthCheckProperty {
+    return CfnTaskDefintion.HealthCheckProperty{
+        Command:     hc.GetCommand(),
+        Interval:    hc.GetInterval(),
+        Retries:     hc.GetRetries(),
+        StartPeriod: hc.GetStartPeriod(),
+        Timeout:     hc.GetTimeout(),
+    }
+}
+```
+
+**NOTE**: To ensure immutability, getters that return a slice type should return a copy of the slice, since elements could be reordered or replaced
+
+It is worth noting that this is not the idiomatic Go way of accessing struct members; normally, a struct would contain unexported members (e.g.
+`command`) and implement a Getter method with the uppercase method name (e.g.  `Command()`). However, having unexported members would make
+instantiating the struct unreasonably difficult for our use case, as struct members can only be set if they are exported. This does make the syntax
+more unwieldy when accessing the members within a function that takes the datatype interface as an argument, but the tradeoff is easier instantiation.
+
+#### Case 2: Extending Typescript datatype interfaces
+
+For Typescript datatype interfaces that extend another datatype interface, the corresponding interfaces and implementing structs would be generated in
+go, with the extending interface embedding the extended one, e.g.:
+
+```ts
+export interface BaseServiceOptions {
+  readonly cluster: ICluster;
+  readonly desiredCount?: number;
+  readonly serviceName?: string;
+  readonly maxHealthyPercent?: number;
+  readonly minHealthyPercent?: number;
+  readonly healthCheckGracePeriod?: Duration;
+  readonly cloudMapOptions?: CloudMapOptions;
+  readonly propagateTags?: PropagatedTagSource;
+  readonly enableECSManagedTags?: boolean;
+  readonly deploymentController?: DeploymentController;
+}
+
+export interface BaseServiceProps extends BaseServiceOptions {
+  readonly launchType: LaunchType;
+}
+```
+
+As before, the Go translation would have the getter methods (e.g. readonly cluster) defined in a the generated interface, which correspond to
+non-exported struct members in the struct that implements each getter.
+
+The interface generated for the jsii struct being extended would be **embedded** in the extending struct.
```

Comment:
> The interface generated for the jsii struct being extended would be **embedded** in the extending struct.

Should this be:

`The interface generated for the jsii struct being extended would be **embedded** in the extending struct`'s interface.